Google Chrome/notifier.js

```diff
@@ -0,0 +1,177 @@
+/*
+* Global variables
+*/
+var _notifCounter = 0;
+var _delayUpdate = 60 * 1000;
+//Used by the popup
+var _currentDom = null;
+var _contentDiv = null;
+//If the user is connected
+var _connected = false;
+//If we want to see popup variables
+var _showNotification = false;
+chrome.storage.local.get('notify', (res) => {
+  _showNotification = res.notify || false;
+});
+//If we are in debug mode
+var _debug = false;
+var _base_url = "https://zestedesavoir.com/";
+var _token = "zds-notifier-firefox";
+if(_debug) _base_url = "https://beta.zestedesavoir.com/";
+
+function escapeHTML(str) { return str.replace(/[&"'<>]/g, (m) => escapeHTML.replacements[m]); }
+escapeHTML.replacements = { "&": "&amp;", '"': "&quot;", "'": "&#39;", "<": "&lt;", ">": "&gt;" };
+
+/**
+* getNotificationsFromAPI
+*/
+function getNotificationsFromAPI() {
+  _contentDiv = document.createElement('div');
+  var target = _base_url + "api/notifications/?page_size=30&ordering=-pubdate&Authorization=" + _token;
+  var xhr = new XMLHttpRequest();
+  xhr.open("GET", target, true);
+  xhr.onload = function (e) {
+    if (xhr.readyState === 4) {
+      var result = xhr.status;
+      if(result === 401) {
+        _connected = false;
+        if(_debug) console.log("Not connected");
+        //Change popup image
+        chrome.browserAction.setIcon({path:"icons/notconnected.png"});
+      } else if (result === 200) {
+        _connected = true;
+        var rootDOM = JSON.parse(xhr.response);
+        if(rootDOM.details) {
+          if(_debug) console.log("Error while parsing");
+        } else {
+          //Get new notifications
+          var resultsNotification = rootDOM.results;
+          var countNotifications = 0;
+          for(var notif = 0; notif < resultsNotification.length; ++notif) {
```

Comment:
Bon ça serait cool de déplacer ça dans une fonction parce qu’il y a des `if` dans des `for` dans des `if` dans des `if` dans des `if`…

Reply:
👍 

À noter qu'un `for (const notif of results)` serait pas mal pour nettoyer le code également.